### DISCUSSION TITLE
segfault when vips_autorot after get orientation

### DISCUSSION THREAD

**Discussion by Flash-nDie** [libvips/libvips#3224] [Help wanted]
hello, i am in trouble when using vips version 8.12.1.
I am newbie in vips.

The following sequence produce segfault when vips_autorot is called.

{
    ...(get image buffer
    VipsImage *in = NULL;
    VipsImage *out = NULL;
    int orient;
    
    vips_image_new_from_buffer(buf, length, "", NULL);
    orient = vips_image_get_orientation(in);
    vips_autorot(in, &out);  -> sig segv occurs
}

without calling 'vips_image_get_orientation', it does work. 
is there something wrong that i misunderstood? 
I tested some images, JPEG, gif, webp..

**Comment by jcupitt** [comment]
Hello @Flash-nDie,

It seems to work for me. You'll need to post a complete program that reproduces the problem before I can help.

**Comment by Flash-nDie** [comment]
@jcupitt 
I just import Vips on working Server Program. 
And I tested with Scoped Function belows.

```
int test_vips(context_t *ctx) {
    VipsImage *in = NULL;
    VipsImage *out = NULL;
    int orient;

    in = vips_image_new_from_buffer(ctx->ori_img_buf, ctx->ori_img_length, "", NULL);
    orient = vips_image_get_orientation(in);
    orient  = vips_autorot(in, &out);

    return ZP_SUCCESS;
}
```

```
int test2_vips(context_t *ctx) {
    VipsImage *in = NULL;
    VipsImage *out = NULL;
    int orient;

    in = vips_image_new_from_buffer(ctx->ori_img_buf, ctx->ori_img_length, "", NULL);

    orient = vips_image_get_orientation(in);
    vips_image_get_width(in);
    vips_image_get_height(in);
    vips_image_get_interpretation(in);
    vips_image_get_n_pages(in);
    orient  = vips_autorot(in, &out);
}

```
I just called test_vips, test2_vips in main(). As mentioned, I import those Two functions in server, So the image buffers are read from fd. 
It does works on image sometimes, occurs segfault sometimes when calling vips_autorot or vips_image_get_orientation. (because of segfault, I run program again and situations are different..) 

I compiled  Vips like belows on Mac:
configure: ./configure --prefix=$prefix --enable-static --disable-shared --without-OpenEXR --without-libjxl --without-libopenjp2 --without-pangocairo

I wanted to build static, and i deleted libtiff because of versions. (I built libtiff directly, So i ran brew uninstall libtiff). 
I think the configure script see the system library path first though i set all values(PKG_CONFIG_PATH, CFLAGS, LDFLAGS and so on). So, in build time, Compiler sees the system libraries. But in runtime, I linked some libraries which built personally. (ex, webp, jpeg, imagemagick, tiff.. ) 
Is this difference between build and run time cause some issues?

**Comment by Flash-nDie** [comment]
@jcupitt
I upgraded version 8.13.3 and i modify vips_autorot(in, &out) -> vips_autorot(in, &out, NULL); 
and it seems work well.
Thank you for managing VIPS and managing it :)

**Reply by jcupitt** [reply]
Ah OK, yes, you need the NULL. If you compile with `-Wall` it'll warn you about things like this.
